python_modules/dagster/dagster/_core/telemetry.py

```diff
@@ -460,12 +460,26 @@ def log_external_repo_stats(
     external_pipeline: Optional["ExternalPipeline"] = None,
 ):
     from dagster._core.host_representation.external import ExternalPipeline, ExternalRepository
+    from dagster._core.host_representation.external_data import (
+        ExternalDynamicPartitionsDefinitionData,
+    )
 
     check.inst_param(instance, "instance", DagsterInstance)
     check.str_param(source, "source")
     check.inst_param(external_repo, "external_repo", ExternalRepository)
     check.opt_inst_param(external_pipeline, "external_pipeline", ExternalPipeline)
 
+    def _get_num_dynamic_partitioned_assets(external_asset_nodes) -> int:
+        return sum(
```

Comment:
I think this could be written slightly more simply as

```python
    sum(
                1
                if asset.partitions_def_data
                and isinstance(asset.partitions_def_data, ExternalDynamicPartitionsDefinitionData)
                for asset in external_asset_nodes
        )